share/models/meta.py

```diff
@@ -47,6 +49,31 @@ def __str__(self):
         return self.url
 
 
+class Subject(models.Model):
+    name = models.TextField(unique=True)
+    lineages = JSONField()
```

Comment:
This should be a ManyToMany field, that we we get referential integrity and it will be much easier to query.

https://github.com/django-mptt/django-mptt/blob/master/mptt/fields.py#L37